cloudferrylib/os/actions/transport_instance.py

```diff
@@ -48,54 +53,124 @@
+        for instance_id, instance in info[utils.INSTANCES_TYPE].iteritems():
             instance = self._replace_user_ids(instance)
 
             one_instance = {
-                utl.INSTANCES_TYPE: {
+                utils.INSTANCES_TYPE: {
                     instance_id: instance
                 }
             }
             one_instance = self.deploy_instance(self.dst_cloud, one_instance)
 
-            new_info[utl.INSTANCES_TYPE].update(
-                one_instance[utl.INSTANCES_TYPE])
+            new_info[utils.INSTANCES_TYPE].update(
+                one_instance[utils.INSTANCES_TYPE])
 
         return {
             'info': new_info
         }
 
-    def deploy_instance(self, dst_cloud, info):
-        info = copy.deepcopy(info)
-        dst_compute = dst_cloud.resources[COMPUTE]
+    def deploy_instance(self, dst_cloud, one_instance):
+        one_instance = copy.deepcopy(one_instance)
+        dst_compute = dst_cloud.resources[utils.COMPUTE_RESOURCE]
+
+        _instance = one_instance[utils.INSTANCES_TYPE].values()[0]
+        instance = _instance[utils.INSTANCE_BODY]
+        new_ids = {}
+        try:
+            new_id = dst_compute.deploy(instance)
+        except exception.TimeoutException:
+            one_instance, new_id = self._deploy_instance_on_random_host(
+                dst_cloud,
+                one_instance
+            )
+
+        new_ids[new_id] = instance['id']
 
-        new_ids = dst_compute.deploy(info)
         new_info = dst_compute.read_info(search_opts={'id': new_ids.keys()})
         for i in new_ids.iterkeys():
             dst_compute.change_status('shutoff', instance_id=i)
         for new_id, old_id in new_ids.iteritems():
             new_instance = new_info['instances'][new_id]
-            old_instance = info['instances'][old_id]
+            old_instance = one_instance['instances'][old_id]
 
             new_instance['old_id'] = old_id
             new_instance['meta'] = old_instance['meta']
             new_instance['meta']['source_status'] = \
                 old_instance['instance']['status']
-            new_instance[utl.INSTANCE_BODY]['key_name'] = \
-                old_instance[utl.INSTANCE_BODY]['key_name']
-        info = self.prepare_ephemeral_drv(info, new_info, new_ids)
-        return info
+            new_instance[utils.INSTANCE_BODY]['key_name'] = \
+                old_instance[utils.INSTANCE_BODY]['key_name']
+        one_instance = self.prepare_ephemeral_drv(
+            one_instance,
+            new_info,
+            new_ids
+        )
+        return one_instance
+
+    def _deploy_instance_on_random_host(self, dst_cloud, one_instance):
+        one_instance = copy.deepcopy(one_instance)
+        dst_compute = dst_cloud.resources[utils.COMPUTE_RESOURCE]
+
+        _instance = one_instance[utils.INSTANCES_TYPE].values()[0]
+        instance_body = _instance[utils.INSTANCE_BODY]
+        az = instance_body['availability_zone']
```

Comment:
This code fails in VM creation in second attempt.
In second attempt, DST availability zone is not replaced so VM creation fails again. 
The code to be used for that shall be like:

    self.attr_override.get_attr(
        instance_body, 'availability_zone',
        default=self.get_availability_zone(
            instance['availability_zone']))